src/NLog/Layouts/JsonLayout.cs

```diff
@@ -76,18 +76,26 @@ protected override string GetFormattedMessage(LogEventInfo logEvent)
             foreach (var col in this.Attributes)
             {
                 jsonWrapper.Inner = col.Layout;
+                jsonWrapper.JsonEncode = col.Encode;
                 string text = jsonWrapper.Render(logEvent);
-                
+
                 if (!string.IsNullOrEmpty(text))
                 {
-                    if (!first) 
+                    if (!first)
                     {
                         sb.Append(", ");
                     }
 
                     first = false;
 
-                    sb.AppendFormat("\"{0}\": \"{1}\"", col.Name, text);
+                    if(col.Encode)
+                    {
+                        sb.AppendFormat("\"{0}\": \"{1}\"", col.Name, text);
```

Comment:
This will enable user to store already formatted JSON in one of the attributes. This code does not encode or additionaly escape attributes that have col.Encode == false. I will refactor for duplication